src/Automation/Automation/Common/AutomationPSClient.cs

```diff
@@ -1657,42 +1657,77 @@ private Azure.Management.Automation.Models.HybridRunbookWorkerGroup TryGetHybrid
         private IDictionary<string, string> ProcessRunbookParameters(string resourceGroupName, string automationAccountName, string runbookName,
             IDictionary parameters)
         {
+            Runbook runbook = null;
+            IEnumerable<KeyValuePair<string, RunbookParameter>> runbookParameters = null;
             parameters = parameters ?? new Dictionary<string, string>();
-            IEnumerable<KeyValuePair<string, RunbookParameter>> runbookParameters =
-                this.ListRunbookParameters(resourceGroupName, automationAccountName, runbookName);
             var filteredParameters = new Dictionary<string, string>();
-
-            foreach (var runbookParameter in runbookParameters)
+            
+            try
             {
-                if (parameters.Contains(runbookParameter.Key))
+                runbook = this.GetRunbook(resourceGroupName, automationAccountName, runbookName);
+            }
+            catch (ResourceCommonException)
+            {
+                // Ignore if runbook does not exists in the account. This is to start global runbooks by name
+                runbookParameters = new Dictionary<string, RunbookParameter>();
+            }
+            
+            if (runbook != null && 0 == String.Compare(runbook.State, RunbookState.New, CultureInfo.InvariantCulture, CompareOptions.IgnoreCase))
```

Comment:
nit: [Yoda conditions](https://en.wikipedia.org/wiki/Yoda_conditions)

Reply:
Thanks for that, that's port of the original code though :)